website/blog/2024-10-21-creating-an-extension.md

```diff
@@ -0,0 +1,74 @@
+---
+title: How to create your first extension
+description: Learn how to create and customize your own extension for Podman Desktop
+slug: extensions-getting-started
+authors: [cdrage]
+tags: [podman-desktop, release, podman]
+hide_table_of_contents: false
+---
+
+![programming](img/creating-an-extension/programming.png)
+
+# How to create your first extension
+
+Extensions are a powerful tool to customize and extend the functionality of Podman Desktop. Whether you want to add new container management features, streamline current workflows, or create custom UI elements specific to your tech stack, building extensions allows you to tailor the Podman Desktop experience to your specific needs. 
+
+In this guide, we'll introduce how you can build your own Podman Desktop extension, with links to detailed documentation that covers each part of the process.
+
+## Introduction to Extensions
+
+Extensions are abundant in Podman Desktop and can be found in the **Extensions -> Catalog** section.
+
+![extension catalog](img/podman-desktop-release-1.10/extension-catalog.png)
+
+Each extension expands on Podman Desktop, such as providing [Kubernetes development clusters with Minikube](https://github.com/containers/podman-desktop-extension-minikube) or even [analyzing your images](https://github.com/containers/podman-desktop-extension-layers-explorer).
+
+Below we will go over how to start creating your first extension!
+
+## Getting Started with Your Project
+
+The first step in creating your extension is setting up the project environment. To learn how to configure the project and add basic components, check out the **[Templates for creating an extension](/docs/extensions/templates)** guide, which walks you through initializing your project from an official template.
+
+## Adding UI Components
```

Comment:
I feel like we are missing a section here something like `architecture` to give some insight of how an extension works, and maybe the difference with the dockers extension which run in containers, where podman extensions are running with node on the host machine.

Reply:
Maybe adding something about the possibility of creating an extension without UI?

Reply:
Created https://github.com/containers/podman-desktop/issues/9593